python/modules/task.py

```diff
@@ -54,29 +53,30 @@ def parse_task_arguments(self, env_local):
             dict_setts[key] = value
         return dict_setts
 
-    def update_dict(self, dict_out, dict_in, list_by_value, list_by_name):
-        for k, v in list_by_value.items(): #update by value
-            dict_out[k] = v
-            logger.debug("Dict_out new key (updated by value): %s", k)
-        for k, v in list_by_name.items(): #update by name
-            logger.debug("Dict_in (by names) key, value: %s, %s", k, v)
-            try:
-                value = dict_in[v]
-            except:
-                logger.error("Dictionary update_by_name error. Given key (%s) doesn't exist in dictionary", v)
-            dict_out[k] = value
-            logger.debug("Dict_out new key, value (updated by name): %s, %s", k, v)
-        logger.debug("Dict_out before concatenation: %s", dict_out.keys())
+    def update_env(self, dict_out, dict_in, variables):
+        for key, var in variables.items():
+            var_class = str(var.__class__.__name__)

```

Comment:
It's not compatible with object-oriented programming  👎  
You should create function get_variable() in class Variable and overload it in VariableList to get the same functionality.
Remember if you need to check what is a class of the object it means that you use OOP improperly.